### PR TITLE
juttle-5-compatible

### PR DESCRIPTION
**Pull request by VladVega**
Goals
- Allow array of credentials. Change option database to db.
- Dynamically change credential object using id option.
- Use adapter API and other Juttle 0.5.0 features.
- Use ES2015 classes.
- Extend from AdapterRead and AdapterWrite.
- Added: Live and super queries.
- run mocha on separate files.
- eslint and test-coverage.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,13 +20,12 @@ before_install:
 
 node_js:
     - '4.2'
-    - '5.0'
 
 before_script:
-    - npm install juttle@>=0.4.0
-    - npm install -g jshint
-    - npm install -g jscs
+    - npm install juttle/juttle;
```
Comment (VladVega) [inline .travis.yml]: temporary.

**Comment by VladVega** [comment]
@demmer @davidvgalbraith

File: .travis.yml (hunk 1 above)
Comment (VladVega) [inline .travis.yml]: will fix now that juttle5 released.

File: lib/read.js [hunk 2]
```diff
@@ -33,107 +38,87 @@ var Read = Juttle.proc.source.extend({
...
-            });
-        }
+    }
+
+    periodicLiveRead() {
+        return true;
+    }
 
+    validateOptions(options) {
         if (!_.has(options, 'name')) {
```
Comment (davidvgalbraith) [inline lib/read.js]: You can put these into `static requiredOptions()`.

File: lib/db.js [hunk 3]
```diff
@@ -1,35 +1,51 @@
...
+'use strict';
+
+/* global JuttleAdapterAPI */
+let logger = JuttleAdapterAPI.getLogger('opentsdb-init');
+
+let _ = require('underscore');
+let opentsdb = require( 'opentsdb');
+let OpentsdbSocket = require('opentsdb-socket');
+let Promise = require('bluebird');
+let client = Promise.promisifyAll(opentsdb.client());
```
Comment (davidvgalbraith) [inline lib/db.js]: Is this single global client safe? Like if I start of one paged/live read from 1 id and then in the middle of that kick off another one from another id won't the first one start reading from the second id?

Comment (VladVega) [inline lib/db.js]: https://github.com/juttle/juttle-opentsdb-adapter/issues/24

File: lib/read.js (hunk 2 above)
Comment (VladVega) [inline lib/read.js]: https://github.com/juttle/juttle-opentsdb-adapter/issues/25

**Comment by davidvgalbraith** [comment]
+1

**Pull request merged**